classes/items/armor.hxx

```diff
@@ -12,23 +12,23 @@
 class Armor: public Item {
 private:
 
-    int defence; //private variable to denote defence
+    int defense; //private variable to denote defence
 
 public:
     Armor(
-            int defence,
-            std::string name,
-            unsigned int tier,
-            unsigned int type,
-            unsigned int value,
-            unsigned int level);
+          std::string name = "Armor",
+          int defense = 0,
+          unsigned int tier = 2,
+          unsigned int type =4,
+          unsigned int value =5,
+          unsigned int level =10);
 
 
 
     virtual ~Armor(); //destructor
 
     //accessors
-    inline const int& get_defence() const { return this->defence; }
+    inline const int& get_defense() const { return this->defense;}
```

Comment:
why are you returning by reference a scalar?

Reply:
I found this to be the easiest way to get it to read into defense. It works properly.

Reply:
the value that's returned is an integer. You're no longer using a pointer at that point. You've already got the property of defense(we can go with the defence spelling if you wish either really it doesn't matter to me). But it's going to be an actual variable and no pointer is there anymore. If we were returning an array or some other struct from within the class then I could see doing something like that.
But since it's a simple scalar it's not needed. That's the only reason I asked b/c it was extraneous code.